testing/regress/ecl/README.md

```diff
@@ -0,0 +1,128 @@
+
+The test suite will evaluate all data types supported by ECL and Arrow, as well as file operations, various compression formats, and schema handling. Additionally, the test suite will measure the plugin's performance across different HPCC components and hardware configurations, conduct stress tests to identify potential bottlenecks and bugs, and compare Parquet to other file formats used in the ecosystem, such as JSON, XML, and CSV.
+
+# Test Files
+The test suite consists of 10 main test files that test different parquet functionality and operations:
+
+- parquet_types.ecl: Tests various ECL and Arrow data types
+
+- parquet_schema.ecl: Evaluates Parquet's handling of different schemas
+
+- parquet_compress.ecl: Tests different compression algorithms
+
+- parquet_write.ecl: Validates Parquet write operations
+
+- parquet_empty.ecl: Tests behavior with empty Parquet files
+
+- parquet_corrupt.ecl: Checks handling of corrupt Parquet data
+
+- parquet_size.ecl: Compares file sizes across formats
+
+- parquet_partition.ecl: Tests partitioning in Parquet files
+
+- parquet_overwrite.ecl: Validates overwrite operations
+
+- parquet-string.ecl: Focuses on string-related operations
+
+
+## Test Suite Overview
+# Type Testing
+
+Covers 42 data types including ECL and Arrow types
+Examples: BOOLEAN, INTEGER, STRING, UNICODE, various numeric types, sets, and Arrow-specific types
+
+ **Data Type Tests**
+
+Tested data types include:
+ECL Types
+- BOOLEAN: True/False values.
+- INTEGER: Signed integer values.
+- UNSIGNED: Unsigned integer values.
+- REAL: Floating-point values.
+- DECIMAL: Decimal values with specified precision.
+- STRING: Fixed-length strings.
+- DATA: Binary data converted to STRING.
+- VARSTRING: Variable-length strings.
+- QSTRING: Quick strings.
+- UTF8: UTF-8 encoded strings.
+- UNICODE: Unicode strings.
+- SET OF INTEGER: Sets of integer values.
+- VARUNICODE: Variable-length Unicode strings.
+- REAL8 (FLOAT8): 8-byte floating-point values.
+- SET OF STRING: Sets of string values.
+- SET OF UNICODE: Sets of Unicode values.
+- INTEGER1 (BYTE): 1-byte integer values.
+
+Arrow Types Supported by the Parquet Plugin
+- null
+- uint8
+- int8
+- uint16
+- int16
+- uint32
+- int32
+- uint64
+- int64
+- half
+- float
+- double
+- string
+- binary
+- fixed_size_binary
+- date32
+- date64
+- timestamp
+- time32
+- time64
+- interval_months
+- list
+- decimal
+- large_list
+- interval_day_time
+
+
+# Compression Testing
+
+Tests all available Arrow compression types: Snappy, GZip, Brotli, LZ4, ZSTD, Uncompressed
+Compares performance and file sizes for different compression options
+
+Parquet Read and Write Operations
+
+Tests ParquetIO.Read for creating ECL datasets from Parquet files
+Tests ParquetIO.Write for writing ECL datasets to Parquet files
+
+# Additional Tests
+
+Spray and despray speeds comparison with other file types
```

Comment:
In this context its better to say read and write speeds. We are also trying to move away from the terminology spray/despray.